### PR TITLE
make network id static

### PR DESCRIPTION
**Pull request by H34D**
## Description

make network id static

## Is this PR related with an open issue?

Related to Issue #291 #290 #296

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
 
## Checklist:

- [ ] Follows the code style of this project.
- [ ] Tests Cover Changes
- [ ] Documentation

#### Funny gif

Put a link of a funny gif inside the parenthesis-->

### CODE REVIEW

**Review by aaitor — APPROVED** [review]

File: integration/nevermined/Artifacts.test.ts [hunk 1]
```diff
@@ -36,7 +36,9 @@ describe('Artifacts', () => {
         writeFileSync(path, buffer)
     }
 
-    tests.forEach(({ nodeUri, networkId, networkName, versions, tag }) => {
+    for(const test of tests) {
```
Comment (aaitor) [inline integration/nevermined/Artifacts.test.ts:39]: lint

File: integration/nevermined/Artifacts.test.ts [hunk 2]
```diff
@@ -36,7 +36,9 @@ describe('Artifacts', () => {
...
+        const { nodeUri, networkId, networkName, versions, tag } = test;
```
Comment (aaitor) [inline integration/nevermined/Artifacts.test.ts:40]: remove `;`

File: src/Instantiable.abstract.ts [hunk 3]
```diff
@@ -30,54 +30,62 @@ export function generateIntantiableConfigFromConfig(
...
-     * Returns network id.
-     * @return {Promise<number>} Network ID.
+     * Returns true of contract exists else it throws.
+     * @return {Promise<boolean>} Contract exists.
      */
-    public async getNetworkId(): Promise<number> {
-        if (!this.networkId) {
-            this.networkId = await this.web3.eth.net.getId()
+    protected async checkExists(address: string): Promise<boolean> {
+        const code = await this.web3.eth.getStorageAt(address, 0);
```
Comment (aaitor) [inline src/Instantiable.abstract.ts:45]: remove `;`

File: src/Instantiable.abstract.ts [hunk 4]
```diff
@@ -30,54 +30,62 @@ export function generateIntantiableConfigFromConfig(
...
+        if (code === '0x0000000000000000000000000000000000000000000000000000000000000000'
+        ) {
+            // no code in the blockchain dude
+            throw new Error(`No contract deployed at address ${address}, sorry.`);
```
Comment (aaitor) [inline src/Instantiable.abstract.ts:49]: same

**Review by aaitor — APPROVED** [review]
lint and style issues but beyond that looks good!

File: integration/nevermined/Artifacts.test.ts (hunk 1 above)
Comment (H34D) [reply integration/nevermined/Artifacts.test.ts]: done

File: integration/nevermined/Artifacts.test.ts (hunk 2 above)
Comment (H34D) [reply integration/nevermined/Artifacts.test.ts]: done

File: src/Instantiable.abstract.ts (hunk 3 above)
Comment (H34D) [reply src/Instantiable.abstract.ts]: done

File: src/Instantiable.abstract.ts (hunk 4 above)
Comment (H34D) [reply src/Instantiable.abstract.ts]: done

**Pull request merged**